### PR TITLE
doc: splits up the release notes more and moves off older items

### PR DESCRIPTION
**Pull request by rnveach**
Documentation update to move off old release notes and add an additional backlog page for the older ones.

### CODE REVIEW

File: config/checkstyle_non_main_files_suppressions.xml [hunk 1]
```diff
@@ -82,6 +82,8 @@
   <!-- content is generated -->
   <suppress id="lineLength"
          files="src[\\/]xdocs[\\/]releasenotes_old.xml"/>
+  <suppress id="lineLength"
+         files="src[\\/]xdocs[\\/]releasenotes_old_2.xml"/>
```
Comment (romani) [inline config/checkstyle_non_main_files_suppressions.xml:86]: Can we

**Review by romani — APPROVED** [review]
With comment

File: config/checkstyle_non_main_files_suppressions.xml (hunk 1 above)
Comment (rnveach) [reply config/checkstyle_non_main_files_suppressions.xml]: Done.

File: src/xdocs/writingchecks.xml [hunk 2]
```diff
@@ -422,7 +422,7 @@ public class MethodLimitCheck extends AbstractCheck
     <section name="Comment Tokens in AST">
 
       <p>
-        Before <a href="releasenotes.html#Release_6.0">Checkstyle 6.0</a>, there was no comments
+        Before <a href="releasenotes_old.html#Release_6.0">Checkstyle 6.0</a>, there was no comments
```
Comment (romani) [inline src/xdocs/writingchecks.xml:425]: Should we apply same naming concept for this file ?

**Review by strkkk — APPROVED** [review]

File: src/xdocs/writingchecks.xml (hunk 2 above)
Comment (rnveach) [reply src/xdocs/writingchecks.xml]: Done.

**Pull request merged**